bot/codeSharingBot.ts

```diff
@@ -0,0 +1,78 @@
+import {
+  TeamsActivityHandler,
+  CardFactory,
+  TurnContext,
+  Attachment,
+} from "botbuilder";
+import { CodeCard } from "./helper/codeCard";
+import { reqCodeDataFromGitHubAPI } from "./helper/reqHelper";
+
+export class CodeSharingBot extends TeamsActivityHandler {
+
+  public async handleTeamsAppBasedLinkQuery(context: TurnContext, query: any): Promise<any> {
+    // Link obtained has `amp;` in the url if the url contains `&`, simply replace it.
+    const url = query.url.replace("amp;","");
+    // Unfurling link contains `github`. 
+    if (url.includes('github.com')){
+      return await handleGitHubUrl(url);
+    }
+  }
+
+  // Using Action as a backup.
+  public async handleTeamsMessagingExtensionSubmitAction(
+    context: TurnContext,
+    action: any
+  ): Promise<any> {
+    switch (action.commandId) {
+      case "createCard":
+        return createCardCommand(context, action);
+      default:
+        throw new Error("NotImplemented");
+    }
+  }
+}
+
+async function createCardCommand(context: TurnContext, action: any): Promise<any> {
```

Comment:
Why do we need to have create card command?

Reply:
A backup command